java-checks/src/main/java/org/sonar/java/checks/IndentationAfterConditionalCheck.java

```diff
@@ -0,0 +1,172 @@
+    super.visitForStatement(tree);
+  }
+
+  @Override
+  public void visitForEachStatement(ForEachStatement tree) {
+    if (thereIsNoIssue(tree)) {
+      return;
+    }
+    report(tree.forKeyword(), tree.closeParenToken(), tree.forKeyword().text(), findSecondariesToReport(tree));
+    super.visitForEachStatement(tree);
+  }
+
+  private static boolean thereIsNoIssue(Tree tree) {
+    return (blockExists(tree) || checkIdentation(tree));
+  }
+
+  public void report(Tree firstTree, Tree secondTree, String operationName, List<JavaFileScannerContext.Location> flow) {
+    context.reportIssue(this, firstTree, secondTree,
+      "Use curly braces or indentation to denote the code conditionally executed by this \"" + operationName + "\".", flow, null);
+  }
+
+  private static boolean checkElseStatementReports(IfStatementTree tree) {
+    if (tree.elseKeyword() != null && !tree.elseStatement().is(Tree.Kind.IF_STATEMENT)) {
+      return tree.elseKeyword().column() >= tree.elseStatement().firstToken().column();
+    }
+    return false;
+  }
+
+  private static List<JavaFileScannerContext.Location> findSecondariesToReport(Tree tree) {
+    Tree parentTree = tree.parent();
+    List<Tree> secondaryLinesToReport = new ArrayList<>();
+    StatementTree statement = returnStatement(tree);
+    secondaryLinesToReport.add(statement);
+    if (parentTree.is(Tree.Kind.BLOCK)) {
+      List<StatementTree> blockStmtList = ((BlockTree) parentTree).body();
+      secondaryLinesToReport.addAll(loopUntilNextEmptyLine(statement.firstToken().line(), blockStmtList, blockStmtList.indexOf(tree)));
+    }
+    return secondaryLinesToReport.stream().map(lineToReport -> new JavaFileScannerContext.Location("", lineToReport))
+      .collect(Collectors.toList());
+  }
+
+  private static List<Tree> loopUntilNextEmptyLine(int previousLine, List<StatementTree> statementsList, int currentLine) {
+    List<Tree> secondaryLinesToReport = new ArrayList<>();
+    for (int i = currentLine + 1; i < statementsList.size(); i++) {
+      StatementTree toBeReportedLine = statementsList.get(i);
+      int currLine = toBeReportedLine.firstToken().line();
+      if (previousLine < currLine - 1
+        && linesBeforeCurrentAreNotCommented(toBeReportedLine.firstToken().trivias().stream().map(SyntaxTrivia::startLine).collect(Collectors.toList()), currLine - 1)) {
+        break;
+      } else {
+        secondaryLinesToReport.add(toBeReportedLine);
+        previousLine = currLine;
+      }
+    }
+    return secondaryLinesToReport;
+
+  }
+
+  private static boolean linesBeforeCurrentAreNotCommented(List<Integer> syntaxTrivia, int prevLine) {
+    return !syntaxTrivia.contains(prevLine);
+  }
+
+  private static int columnStatement(Tree tree) {
+    return tree.firstToken().column();
+  }
+
+  private static boolean checkIdentation(Tree tree) {
+    return returnStatement(tree).firstToken().column() > columnStatement(tree);
```

Comment:
You can also use invocation to `columnStatement` in the first part of the condition